src/lib/mina_base/test/account/account.ml

```diff
@@ -0,0 +1,273 @@
+open Core_kernel
+open Currency
+open Mina_base
+open Mina_numbers
+open Unsigned
+
+let gen_timing = Account.gen_timing Balance.max_int
+
+let gen_timing_with_necessary_vesting =
+  let open Quickcheck in
+  let open Account.Timing.As_record in
+  Account.gen_timing Balance.max_int
+  |> Generator.filter ~f:(fun t ->
```

Comment:
You should patch up the `Account.gen_timing` generator instead of doing this. This can cause the generator to take an unnecessarily long amount of time. If we need to still let the `Account.gen_timing` generator generate timings where `initial_minimum_balance <= cliff_amount`, we can add a flag to the generator to toggle that behavior.